openedx/features/course_experience/views/course_home_messages.py

```diff
@@ -124,7 +127,8 @@ def _register_course_home_messages(request, course, user_access, course_start_da
             ),
             title=Text(_('You must be enrolled in the course to see course content.'))
         )
-    if not user_access['is_anonymous'] and not user_access['is_staff'] and not user_access['is_enrolled']:
+    if not user_access['is_anonymous'] and not user_access['is_staff'] and \
+            not user_access['is_enrolled'] and not course.invitation_only:
         CourseHomeMessages.register_info_message(
```

Comment:
There needs to be a message displayed when the user is unenrolled and course is set to invitation_only. You can display something like `You should be enrolled in this course to have full access to the contents`.

Reply:
I've added the following if-condition:
   ```
 if not user_access['is_enrolled'] and course.invitation_only:
         CourseHomeMessages.register_info_message(
              request,
               Text(_(
                   u'{open_enroll_link}Enroll now{close_enroll_link} to access the full course.'
               )).format(
                   open_enroll_link=HTML('<button class="enroll-btn btn-link">'),
                   close_enroll_link=HTML('</button>')
               ),
               title=Text(_(u'Welcome to {course_display_name}')).format(
                   course_display_name=course.display_name
              )
          )
 
```

Reply:
Could you clarify, for what kind of course? My understanding is that we should not show enroll message unless self enrollment is allowed.

Reply:
When the user is logged in but not enrolled and the course is set to invitation_only (i.e. self enrollment is not allowed), it would be good to show a message to the user as there could be some sections in the course that may be only accessible to enrolled students (even though the course is marked public).

Reply:
Ok sure thing.